File: 219-ContainsDuplicateII/219-ContainsDuplicateII.java
// Last updated: 8/11/2026, 9:22:38 AM
1public class Solution {
2    public int longestPalindrome(String s) {
3        // Initialize a set to keep track of characters with odd frequencies
4        HashSet<Character> charSet = new HashSet<>();
5        // Initialize the length of the longest palindrome
6        int length = 0;
7        
8        // Iterate over each character in the string
9        for (char c : s.toCharArray()) {
10            // If the character is already in the set, remove it and increase the length by 2
11            if (charSet.contains(c)) {
12                charSet.remove(c);
13                length += 2;
14            } else {
15                // If the character is not in the set, add it to the set
16                charSet.add(c);
17            }
18        }
19        
20        // If there are any characters left in the set, add 1 to the length for the middle character
21        if (!charSet.isEmpty()) {
22            length += 1;
23        }
24        
25        // Return the total length of the longest palindrome
26        return length;
27    }
28}

File: 219-ContainsDuplicateII/219-ContainsDuplicateII.py
# Last updated: 8/11/2026, 9:02:35 AM
1class Solution:
2    def wordPattern(self, p: str, s: str) -> bool:
3        p = list(p)
4        s = s.split(' ')
5
6        if len(p) != len(s):
7            return False
8
9        h = {}
10
11        for i in range(len(p)):
12            if p[i] in h and s[i] != h[p[i]]:
13                return False
14
15            elif p[i] not in h and s[i] in h.values():
16                return False
17
18            else:
19                h[p[i]] = s[i]
20
21        return True